AnkiDroid/src/test/java/com/ichi2/preferences/TimePreferenceTest.java

```diff
@@ -0,0 +1,39 @@
+package com.ichi2.preferences;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(value = Parameterized.class)
+public class TimePreferenceTest {
+    private final String parsableHour;
+    private final int expectedHour;
+
+
+    public TimePreferenceTest(String parsableHour, int expectedHour) {
+        this.parsableHour = parsableHour;
+        this.expectedHour = expectedHour;
+    }
+
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                {"00:00", 0},
+                {"01:00", 1},
+                {"24:00", 24}, // should this be "0"?
```

Comment:
Yes, we should be doing this mod 24. 

Wanna comment it out and put in an issue (or fix it) so we can deal with it later and get this review in

Reply:
I'll create an issue so the appropriate behavior is defined there.

```suggestion
                {"24:00", 24}
```